src/psyclone/psyir/nodes/reference.py

```diff
@@ -68,15 +68,15 @@ def __eq__(self, other):
         '''
         Checks equivalence of two References. References are considered
         equivalent if they are the same type of Reference and their symbol
-        is the same.
+        name is the same.
 
         :param object other: the object to check equality to.
 
         :returns: whether other is equal to self.
         :rtype: bool
         '''
         is_eq = super().__eq__(other)
-        is_eq = is_eq and (self.symbol == other.symbol)
+        is_eq = is_eq and (self.symbol.name == other.symbol.name)
```

Comment:
The practical reason is that the symbol equality is not implemented yet. So before this was only testing if the object was the same id in memory and it returned False for symbols which where exactly equivalent.

But there is more, in:
```
subroutine a()
    real :: b
    b = b + 1
end subroutine
subroutine a()
    integer :: b
    b = b + 1
end subroutine
```
The equality of comparing both subroutines must return False (because their declaration is different).

But does the equality between the addition statements should be True or False? There is no right/wrong answer, it is what we decide for a reference to be equal means. (maybe I should raise a different issue to discuss this.)

Reply:
Added a TODO